qiskit/circuit/quantumcircuit.py

```diff
@@ -1262,31 +1263,46 @@ def parameters(self):
         """Convenience function to get the parameters defined in the parameter table."""
         return set(self._parameter_table.keys())
 
-    def bind_parameters(self, value_dict):
+    @property
+    def num_parameters(self):
+        """Convenience function to get the number of parameter objects in the circuit."""
+        return len(self.parameters)
+
+    def bind_parameters(self, value_dict, in_place=False):
         """Assign parameters to values yielding a new circuit.
 
         Args:
             value_dict (dict): {parameter: value, ...}
+            in_place (bool): If False, a copy of the circuit with the bound parameters is
+                returned, if True the circuit instance itself is modified.
 
         Raises:
             CircuitError: If value_dict contains parameters not present in the circuit
 
         Returns:
-            QuantumCircuit: copy of self with assignment substitution.
+            QuantumCircuit: The circuit or a copy of it (depending on the value of ``in_place``)
+                with assignment substitution.
```

Comment:
If `inplace=True`, we should return `None` (for clarity to users).

Reply:
Yes, I thought about that but wasn't sure if it is bad style to return `Optional[QuantumCircuit]`. I'll change it.